Readability.js

```diff
@@ -244,6 +247,21 @@ Readability.prototype = {
     Array.prototype.forEach.call(nodeList, fn, this);
   },
 
+  /**
+   * Iterate over a NodeList, and return the first node that passes
+   * the supplied test function
+   *
+   * For convenience, the current object context is applied to the provided
+   * test function.
+   *
+   * @param  NodeList nodeList The NodeList.
+   * @param  Function fn       The test function.
+   * @return void
+   */
+  _findNode: function(nodeList, fn) {
```

Comment:
Added support for yet another array method on NodeLists. If we're okay with the browser support for [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) (which I notice is already in the codebase), we could just cast all NodeLists to arrays and renounce these wrapper functions altogether.

Reply:
Hm, that's an interesting one - my understanding is that `Array.from` will create a copy, which IMO is only really worth it when you know you're going to modify the contents of the list (e.g. by removing elements from the DOM) which messes up live nodelist iteration otherwise. We're not doing that here, so I'd prefer to avoid the copy for now.

I do agree that it causes clutter to have so many wrappers... perhaps there's a way to clean this up in a separate PR; we could e.g. dynamically create these wrappers from a list of methods we're interested in using. Something vaguely like:

```js
for (let method of ["find", "forEach", "some", "every"]) {
  Readability.prototype["_" + method + "Node"] = function(nodelist, fn) {
    return Array.prototype[method].call(nodeList, fn, this);
  };
}
```

or something.

Reply:
I don't have data nor an intuition on the performance tradeoff of `Array.from` vs. `call()`-ing array methods on the NodeList, but I agree a separate effort/PR is the way to go here.